open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/DefaultConsentValidator.java

```diff
@@ -381,4 +383,116 @@ private static void validateFundsConfirmationSubmission(ConsentValidateData cons
         consentValidationResult.setValid(true);
 
     }
+
+    /**
+     * Validate VRP Submission Request.
+     *
+     * @param consentValidateData     Object with request data
+     * @param consentValidationResult Validation result object to return
+     */
+    private void validateVRPSubmission(ConsentValidateData consentValidateData, JSONObject initiationJson,
+                                       ConsentValidationResult consentValidationResult) {
+
+        DetailedConsentResource detailedConsentResource = consentValidateData.getComprehensiveConsent();
+
+        if (!ConsentExtensionConstants.AUTHORIZED_STATUS
+                .equals(consentValidateData.getComprehensiveConsent().getCurrentStatus())) {
+            log.error(ErrorConstants.PAYMENT_CONSENT_STATE_INVALID);
+            consentValidationResult.setErrorMessage(ErrorConstants.PAYMENT_CONSENT_STATE_INVALID);
+            consentValidationResult.setErrorCode(ErrorConstants.RESOURCE_INVALID_CONSENT_STATUS);
+            consentValidationResult.setHttpCode(HttpStatus.SC_BAD_REQUEST);
+            return;
+        }
+
+        // Check if requested consent ID matches to initiation consent ID.
+        if (consentValidateData.getConsentId() == null || detailedConsentResource.getConsentID() == null ||
+                !consentValidateData.getConsentId().equals(detailedConsentResource.getConsentID())) {
+            log.error(ErrorConstants.MSG_INVALID_CONSENT_ID);
+            ConsentValidatorUtil.getValidationResult(ErrorConstants.RESOURCE_CONSENT_MISMATCH,
+                    ErrorConstants.MSG_INVALID_CONSENT_ID);
+            return;
+        }
+
+        JSONObject submissionJson = consentValidateData.getPayload();
+        JSONObject submissionData = (JSONObject) submissionJson.get(ConsentExtensionConstants.DATA);
```

Comment:
Let's move this line to L425 , since this is used in L427

Reply:
Fixed in [8362873]